shared/middleware/files_api.rb

```diff
@@ -469,7 +469,7 @@ def copy_file(endpoint, encrypted_channel_id, filename, source_filename)
     content_type :json
 
     bucket = FileBucket.new
-    result = bucket.get(encrypted_channel_id, FileBucket::MANIFEST_FILENAME, env['HTTP_IF_MODIFIED_SINCE'])
+    result = bucket.get(encrypted_channel_id, FileBucket::MANIFEST_FILENAME, env['HTTP_IF_MODIFIED_SINCE'], params['version'])
```

Comment:
I think `test_files.rb` should have a new test case that checks that this `version` parameter produces the expected result.